### PR TITLE
feat(auto-import-list): Implementing auto import list

### LINKED ISSUE #104: Implement flow for managing existing auto-imports
image

### PR DESCRIPTION
**Pull request by german-mergestat**
Implementing auto import list. (Resolves #104)

- Listing auto imports
- Unit testing

manage-imports

### CODE REVIEW

File: ui/src/api-logic/mappers/imports.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { RepoImportData } from 'src/@types'
+import { SYNC_REPO_METHOD } from 'src/utils/constants'
+import { GetRepoImportsQuery } from '../graphql/generated/schema'
+
+/**
+ * Method which iterate each repo import and map it to RepoImportData to be shown in table
+ * @param data Repo import list that comes from data base in GetRepoImportsQuery format
+ * @returns Repo import list from data base mapped to RepoImportData list
+ */
+const mapToImpotsData = (data: GetRepoImportsQuery | undefined): Array<RepoImportData> => {
```
Comment (patrickdevivo) [inline ui/src/api-logic/mappers/imports.ts:10]:
```suggestion
const mapToImportsData = (data: GetRepoImportsQuery | undefined): Array<RepoImportData> => {
```

Seems like a missing `r` typo?

File: ui/src/views/repositories/hooks/useRepoImports.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const useRepoImports = () => {
+  const [imports, setImports] = useState<RepoImportData[]>([])
+
+  const { loading, error, data, refetch } = useQuery<GetRepoImportsQuery>(GET_REPO_IMPORTS, {
+    pollInterval: 5000,
+  })
+
+  useEffect(() => {
+    setImports(mapToImpotsData(data))
```
Comment (patrickdevivo) [inline ui/src/views/repositories/hooks/useRepoImports.ts:17]:
```suggestion
    setImports(mapToImportsData(data))
```
see above

File: ui/src/api-logic/mappers/imports.ts (hunk 1 above)
Comment (german-mergestat) [reply ui/src/api-logic/mappers/imports.ts]: Done [414f4b3](https://github.com/mergestat/fuse/pull/200/commits/414f4b30543eda7033cd30d4eded4b4df6c21afc)

File: ui/src/views/repositories/hooks/useRepoImports.ts (hunk 2 above)
Comment (german-mergestat) [reply ui/src/views/repositories/hooks/useRepoImports.ts]: Done [414f4b3](https://github.com/mergestat/fuse/pull/200/commits/414f4b30543eda7033cd30d4eded4b4df6c21afc)

**Review by patrickdevivo — DISMISSED** [review]

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**